### ISSUE TITLE
pdfreport can't work when credentials are required

### ISSUE THREAD

**Issue by ger-benjamin** [camptocamp/c2cgeoportal#3041]
Since GMF 2.1

The parameters given to these methods: https://github.com/camptocamp/c2cgeoportal/blob/2.1/c2cgeoportal/views/pdfreport.py#L140-L141

Doesn't match with the signature of this method: https://github.com/camptocamp/c2cgeoportal/blob/2.1/c2cgeoportal/lib/filter_capabilities.py#L57 , respectively this method: https://github.com/camptocamp/c2cgeoportal/blob/2.1/c2cgeoportal/lib/filter_capabilities.py#L63

**Comment by sbrunner** [comment]
=>
in init:
```
self.mapserver_settings = request.registry.settings.get("mapserverproxy", {})
if "default_ogc_server" in self.mapserver_settings:
    self.default_ogc_server = self.get_ogcserver_byname(
        self.mapserver_settings["default_ogc_server"]
)
```
and
```
if self.layername in get_private_layers(self.default_ogc_server) and \
    self.layername not in get_protected_layers(role_id, self.default_ogc_server):
```
:-)

**Comment by ger-benjamin** [comment]
`self.layername in get_private_layers(self.default_ogc_server)`

What if our layer is not in the default_ogc_server ?

**Comment by ger-benjamin** [comment]
Closed with #3047

**Comment by ger-benjamin** [comment]
Reopen because names are compared to object...
